src/sage/rings/number_field/order.py

```diff
@@ -119,6 +119,83 @@ def quadratic_order_class_number(disc):
         h = pari.qfbclassno(disc)
     return ZZ(h)
 
+def quadratic_order_approximate_class_number(disc, *, bound=10**4):
```

Comment:
putting it here is rather undiscoverable though. How about making it (also) a member of `QuadraticField`?

Reply:
I was even tempted to add it as an `algorithm=` option to the existing `.class_number()` method, but that seems questionable. Perhaps it would be enough to mention the existence of this function in the documentation of `.class_number()`? It is a rather obscure feature that most users will never require, after all...

Reply:
mention in the documentation is fine by me.

algorithm= is indeed questionable, since the function does an entirely different things.

Also the lint error should be fixed